src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp

```diff
@@ -147,31 +146,19 @@ class boss_mother_shahraz : public CreatureScript
             Talk(SAY_DEATH);
         }
 
-        void TeleportPlayers()
+        void EnterEvadeMode(EvadeReason /*why*/) override
         {
-            uint32 random = urand(0, 6);
-            float X = TeleportPoint[random].x;
-            float Y = TeleportPoint[random].y;
-            float Z = TeleportPoint[random].z;
-            for (uint8 i = 0; i < 3; ++i)
-            {
-                if (Unit* unit = SelectTarget(SELECT_TARGET_RANDOM, 1))
-                    if (unit->IsAlive() && unit->GetTypeId() == TYPEID_PLAYER)
-                    {
-                        TargetGUID[i] = unit->GetGUID();
-                        unit->CastSpell(unit, SPELL_TELEPORT_VISUAL, true);
-                        DoTeleportPlayer(unit, X, Y, Z, unit->GetOrientation());
-                    }
-            }
+            _DespawnAtEvade();
         }
 
         void DamageTaken(Unit* /*attacker*/, uint32 &damage) override
         {
-            if (!Enraged && me->HealthBelowPctDamaged(10, damage))
+            if (!_enraged && me->HealthBelowPctDamaged(10, damage))
             {
-                Enraged = true;
-                DoCast(me, SPELL_ENRAGE, true);
+                _enraged = true;
+                DoCastSelf(SPELL_RANDOM_PERIODIC, true);
                 Talk(SAY_ENRAGE);
+                Talk(EMOTE_ENRAGE, me);
```

Comment:
hmm for me this "boss warning" in middle of screen is emote haha xd
i will rename it